File: programmers/kakao/level2/문자열_압축.py
def solution(s):
    answer = len(s)
    for x in range(1, len(s)//2+1):
        d = 0
        comp = ""
        c = 1
        for i in range(0, len(s), x):
            tmp = s[i:i+x]
            if comp == tmp:
                c += 1
            elif comp != tmp:
                d += len(tmp)
                if c > 1:
                    d += len(str(c))
                c = 1
                comp = tmp
        if c > 1:
            d += len(str(c))
        answer = min(answer, d)
    return answer


s = "aabbaccc"
print(solution(s)) # 7

s = "ababcdcdababcdcd"
print(solution(s)) # 9

s = "abcabcdede"
print(solution(s)) # 8

s = "abcabcabcabcdededededede"
print(solution(s)) # 14

s = "xababcdcdababcdcd"
print(solution(s)) # 17

File: Java/programmers/kakao/level2/문자열압축.java
package CodeTest.Java.programmers.kakao.level2;

public class 문자열압축 {
    public int solution(String s) {
        int answer = s.length();
        for (int x=1; x<s.length()/2+1; x++) {
            int d = 0;
            String comp = "";
            int c = 1;
            
            for (int i=0; i<s.length(); i+=x) {
                String tmp = "";
                if (i+x <= s.length()) {
                    tmp = s.substring(i, i+x);
                } else {
                    tmp = s.substring(i);
                }
                
                if (comp.equals(tmp)) c++;
                else {
                    d += tmp.length();
                    if (c > 1) {
                        d += String.valueOf(c).length();
                    }
                    c = 1;
                    comp = tmp;
                }
            }
            
            if (c > 1) {
                d += String.valueOf(c).length();
            }
            answer = Math.min(answer, d);
        }
        return answer;
    }
}
